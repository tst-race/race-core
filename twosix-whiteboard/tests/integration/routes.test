#!/bin/bash

DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
source "${DIR}"/helpers.sh

start_containers

# TODO health check?
sleep 2

echo "get latest index for non-existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/latest/foo" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/latest/bar" | jq -c  'del(.timestamp)'

echo "get index for non-existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1" | jq -c  'del(.timestamp)'

echo "get range for non-existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/-2" | jq -c  'del(.timestamp)'

echo "post to previously non-existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request POST "localhost:5000/post/foo" --header 'Content-Type: application/json' --data-raw '{ "data":"0" }' | jq -c  'del(.timestamp)'

echo "get latest index for existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/latest/foo" | jq -c  'del(.timestamp)'

echo "get latest index for non-existent category when another category exists" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/latest/bar" | jq -c  'del(.timestamp)'

echo "get index for existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1" | jq -c  'del(.timestamp)'

echo "get index for non-existent category when another category exists" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1" | jq -c  'del(.timestamp)'

echo "get range for existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/-1" | jq -c  'del(.timestamp)'

echo "get range for non-existent category when another category exists" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/-1" | jq -c  'del(.timestamp)'

echo "post to existent category" | tee >(cat >&2)
curl --fail --silent --show-error --request POST "localhost:5000/post/foo" --header 'Content-Type: application/json' --data-raw '{ "data":"1" }' | jq -c  'del(.timestamp)'

echo "get latest index for existent category (2)" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/latest/foo" | jq -c  'del(.timestamp)'

echo "get index for existent category (2)" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1" | jq -c  'del(.timestamp)'

echo "get range for existent category (2)" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/-1" | jq -c  'del(.timestamp)'

echo "post to second category" | tee >(cat >&2)
curl --fail --silent --show-error --request POST "localhost:5000/post/bar" --header 'Content-Type: application/json' --data-raw '{ "data":"bar0" }' | jq -c  'del(.timestamp)'

echo "get latest index for first category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/latest/foo" | jq -c  'del(.timestamp)'

echo "get index for first category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1" | jq -c  'del(.timestamp)'

echo "get range for first category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/1/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/2/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-1/-1" | jq -c  'del(.timestamp)'

echo "get latest index for second category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/latest/bar" | jq -c  'del(.timestamp)'

echo "get index for second category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1" | jq -c  'del(.timestamp)'

echo "get range for second category" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/0/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/1/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/2/-1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/1" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/2" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/bar/-1/-1" | jq -c  'del(.timestamp)'

echo "latest bad urls" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/latest/" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/latest//" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/latest/foo/~/" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/latest/foo/~/~" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/latest/foo/0/0" | jq -c  'del(.timestamp)'

echo "get index bad urls" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get/" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get//" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo//" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/123456789012345678901234567890" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-123456789012345678901234567890" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/why?" | jq -c  'del(.timestamp)'

echo "get range bad urls" | tee >(cat >&2)
curl --fail --silent --show-error --request GET "localhost:5000/get///" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo//" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/" | jq -c  'del(.timestamp)'
# curl --fail --silent --show-error --request GET "localhost:5000/get/foo//0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/~/" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/~/~" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/~/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/a/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/~" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/a" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/123456789012345678901234567890/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/123456789012345678901234567890" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/-123456789012345678901234567890/0" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/0/-123456789012345678901234567890" | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request GET "localhost:5000/get/foo/why/not?" | jq -c  'del(.timestamp)'

echo "post bad urls" | tee >(cat >&2)
curl --fail --silent --show-error --request POST "localhost:5000/post/" --header 'Content-Type: application/json' --data-raw '{ "data":"bar0" }' | jq -c  'del(.timestamp)'
curl --fail --silent --show-error --request POST "localhost:5000/post/a/b" --header 'Content-Type: application/json' --data-raw '{ "data":"bar0" }' | jq -c  'del(.timestamp)'

stop_containers