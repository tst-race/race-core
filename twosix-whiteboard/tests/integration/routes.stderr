building docker image in case it's not up-to-date
create docker network if in case it doesn't exist
start containers
get latest index for non-existent category
get index for non-existent category
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
get range for non-existent category
post to previously non-existent category
get latest index for existent category
get latest index for non-existent category when another category exists
get index for existent category
curl: (22) The requested URL returned error: 404 NOT FOUND
get index for non-existent category when another category exists
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
get range for existent category
get range for non-existent category when another category exists
post to existent category
get latest index for existent category (2)
get index for existent category (2)
curl: (22) The requested URL returned error: 404 NOT FOUND
get range for existent category (2)
post to second category
get latest index for first category
get index for first category
curl: (22) The requested URL returned error: 404 NOT FOUND
get range for first category
get latest index for second category
get index for second category
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
get range for second category
latest bad urls
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
get index bad urls
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
get range bad urls
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
post bad urls
curl: (22) The requested URL returned error: 404 NOT FOUND
curl: (22) The requested URL returned error: 404 NOT FOUND
stop containers
